BUILD-353: name changes, 2 CRDs instead of one for secrets and configmaps

gabemontero opened the pull request:
in repro API should be drag and drop back to openshift/api

the 2 CRDs resulted in removing support for switching between configmaps and secrets for a "share" (a good thing IMO in hindsight)

/assign @coreydaley 
/assign @adambkaplan

gabemontero commented:
we have green e2e's on the latest rename @coreydaley @adambkaplan

coreydaley commented on docs/content-update-details.md:
```diff
@@ -42,6 +42,7 @@ provided by this CSI driver.
 Allowing the disabling processing of updates, or switching the default for the system as not dealing with
 updates, but then allowing for opting into updates, is also under consideration.
 
+//TODO do we punt on this with the separate CRDs now
```
I think this can be removed.

gabemontero replied:
yep thanks

coreydaley commented on docs/csi.md:
```diff
@@ -3,13 +3,13 @@
 
 - for the `Driver` string field, it needs to be ["csi-driver-projected-resource.openshift.io"](https://github.com/openshift/csi-driver-shared-resource/blob/1fcc354faa31f624086265ea2228661a0fc2e7b1/pkg/client/client.go#L28).
-- for the `VolumeAttributes` map, this driver currently adds the "share" key (which maps the the `SharedResource` instance your `Pod` wants to use) in addition to the
+- for the `VolumeAttributes` map, this driver currently adds the "share" key (which maps the the `SharedConfigMap` OR `SharedSecret` instance your `Pod` wants to use) in addition to the
```
```suggestion
- for the `VolumeAttributes` map, this driver currently adds the "share" key (which maps the `SharedConfigMap` OR `SharedSecret` instance your `Pod` wants to use) in addition to the
```

gabemontero replied:
yeah actually the key name is not correct as well

it is sharedConfigMap or sharedSecret

I'll make that changes as well here

coreydaley commented on docs/install.md:
```diff
@@ -46,9 +47,13 @@ Kubernetes resources:
+oc apply -f ./deploy/0000_10_sharedconfigmap.crd.yaml
+customresourcedefinition.apiextensions.k8s.io/sharedconfigmaps.sharedresource.openshift.io created
+deploying hostpath components
+   ./deploy/0000_10_sharedsecrets.crd.yaml
```
```suggestion
   ./deploy/0000_10_sharedsecret.crd.yaml
```

coreydaley commented on docs/install.md:
```diff
@@ -107,26 +114,28 @@ daemonset.apps/csi-hostpathplugin created
-oc apply -f --filename https://raw.githubusercontent.com/openshift/csi-driver-shared-resource/release-4.8/deploy/03-cluster-role-binding.yaml
-oc apply -f --filename https://raw.githubusercontent.com/openshift/csi-driver-shared-resource/release-4.8/deploy/csi-hostpath-driverinfo.yaml
-oc apply -f --filename https://raw.githubusercontent.com/openshift/csi-driver-shared-resource/release-4.8/deploy/csi-hostpath-plugin.yaml 
+oc apply -f --filename https://raw.githubusercontent.com/openshift/csi-driver-shared-resource/release-4.10/deploy/00-namespace.yaml
+oc apply -f --filename https://raw.githubusercontent.com/openshift/csi-driver-shared-resource/release-4.10/deploy/0000_10_sharedconfigmap.crd.yaml
+oc apply -f --filename https://raw.githubusercontent.com/openshift/csi-driver-shared-resource/release-4.10/deploy/0000_10_sharedsecret.crd.yaml
+oc apply -f --filename https://raw.githubusercontent.com/openshift/csi-driver-shared-resource/release-4.10/deploy/01-service-account.yaml 
+oc apply -f --filename https://raw.githubusercontent.com/openshift/csi-driver-shared-resource/release-4.10/deploy/02-cluster-role.yaml
+oc apply -f --filename https://raw.githubusercontent.com/openshift/csi-driver-shared-resource/release-4.10/deploy/03-cluster-role-binding.yaml
+oc apply -f --filename https://raw.githubusercontent.com/openshift/csi-driver-shared-resource/release-4.10/deploy/csi-hostpath-driverinfo.yaml
+oc apply -f --filename https://raw.githubusercontent.com/openshift/csi-driver-shared-resource/release-4.10/deploy/csi-hostpath-plugin.yaml 
```
Should we replace `release-4.10` here with `master` instead?

gabemontero replied:
I don't think so, as master could be unstable, once a given release goes out.

coreydaley replied:
So we will have to update this readme every release?

gabemontero replied:
lesser of the evils IMO ... at least for now ... and if it lags, it is not the worse thing ... and it is easy enough for someone to figure our they can tweak that to another branch

possible this just goes away or morphs into a "dev only" mode at some point, in which case we switch to master 

I'd rather hold off on that while we are in this intermediate state

coreydaley commented on pkg/api/sharedresource/v1alpha1/share_configmap_type.go:
```diff
@@ -0,0 +1,73 @@
+//
+// spec:
+//  volumes:
+//  - name: shared-secret
```
```suggestion
//  - name: shared-configmap
```

coreydaley commented on pkg/api/sharedresource/v1alpha1/share_configmap_type.go:
```diff
@@ -0,0 +1,73 @@
+package v1alpha1
```
Should this file be named `shared_configmap_type.go` ?

gabemontero replied:
yeah and the secret one too

gabemontero commented:
pushed updates from your comments as separate commits @coreydaley - thanks !

coreydaley commented on cmd/main.go:
```diff
@@ -79,7 +78,9 @@ func init() {
 	rootCmd.Flags().AddGoFlagSet(flag.CommandLine)
 	rootCmd.Flags().StringVar(&endPoint, "endpoint", "unix://tmp/csi.sock", "CSI endpoint")
-	rootCmd.Flags().StringVar(&driverName, "drivername", string(v1.SharedResourcesCSIDriver), "name of the driver")
+	//rootCmd.Flags().StringVar(&driverName, "drivername", string(v1.SharedResourcesCSIDriver), "name of the driver")
```
Do we still need this commented out line?

gabemontero replied:
I'd like to leave it as per the TODO comment below we'll swtich back to it once you've updated that driver name with your openshift/api PR

coreydaley commented on docs/simple-example.md:
```diff
@@ -97,7 +92,7 @@ $ kubectl apply -f ./examples
 role.rbac.authorization.k8s.io/shared-resource-my-share created
 rolebinding.rbac.authorization.k8s.io/shared-resource-my-share created
-share.sharedresource.storage.openshift.io/my-share created
+sharedconfig.sharedresource.openshift.io/my-share created
```
Should this be `sharedconfigmap` ?

gabemontero replied:
yep

coreydaley commented on docs/faq.md:
````diff
@@ -1,20 +1,20 @@
 ```bash
 $ oc get events
-0s          Warning   FailedMount      pod/my-csi-app                                       MountVolume.SetUp failed for volume "my-csi-volume" : rpc error: code = InvalidArgument desc = the csi driver volumeAttribute 'share' reference had an error: sharedresource.storage.openshift.io "my-share" not found
+0s          Warning   FailedMount      pod/my-csi-app                                       MountVolume.SetUp failed for volume "my-csi-volume" : rpc error: code = InvalidArgument desc = the csi driver volumeAttribute 'share' reference had an error: sharedconfig.sharedresource.openshift.io "my-share" not found
````
Should this be `sharedconfigmap`?

gabemontero replied:
yep

coreydaley commented on pkg/api/sharedresource/v1alpha1/0000_10_sharedsecret.crd.yaml:
```diff
@@ -0,0 +1,22 @@
+# this is the boilerplate crd def that controller-gen reads and modifies with the
+# contents from share_secret_type.go
```
```suggestion
# contents from shared_secret_type.go
```

gabemontero commented:
ok @coreydaley squashed the previous review commits, add a new one for the next round of corrections from you - thanks !

coreydaley commented on deploy/0000_10_sharedconfigmap.crd.yaml:
```diff
@@ -0,0 +1,104 @@
+# this is the boilerplate crd def that controller-gen reads and modifies with the
+# contents from share_configmap_type.go
```
```suggestion
# contents from shared_configmap_type.go
```

coreydaley commented on deploy/0000_10_sharedsecret.crd.yaml:
```diff
@@ -1,27 +1,27 @@
 # this is the boilerplate crd def that controller-gen reads and modifies with the
-# contents from share_type.go
+# contents from share_secret_type.go
```
```suggestion
# contents from shared_secret_type.go
```

coreydaley commented on pkg/api/sharedresource/v1alpha1/shared_secret_type.go:
```diff
@@ -0,0 +1,74 @@
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SharedSecret allows a Secret o be shared across namespaces.
```
```suggestion
// SharedSecret allows a Secret to be shared across namespaces.
```

coreydaley commented on pkg/api/sharedresource/v1alpha1/shared_configmap_type.go:
```diff
@@ -0,0 +1,73 @@
+//        sharedConfigMap: my-share
+//
+// For the mount to be successful, the pod's service account must be granted permission to 'use' the named SharedConfigMap object
+// within its namespace with an appropriate Role and RoleBinding.  For compactness, here are example `oc` invocations for creating
```
```suggestion
// within its namespace with an appropriate Role and RoleBinding. For compactness, here are example `oc` invocations for creating
```

coreydaley commented on pkg/api/sharedresource/v1alpha1/shared_configmap_type.go:
```diff
@@ -0,0 +1,73 @@
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SharedConfigMapList contains a list of ShareConfigMap objects.
```
```suggestion
// SharedConfigMapList contains a list of SharedConfigMap objects.
```

coreydaley commented on pkg/api/sharedresource/v1alpha1/shared_configmap_type.go:
```diff
@@ -0,0 +1,73 @@
+// SharedConfigMapSpec defines the desired state of a SharedConfigMap
+// +k8s:openapi-gen=true
+type SharedConfigMapSpec struct {
+	// resource references the backing ConfigMap for this SharedConfigMap.
```
```suggestion
	// configMap references the backing ConfigMap for this SharedConfigMap.
```

coreydaley commented on pkg/api/sharedresource/v1alpha1/shared_secret_type.go:
```diff
@@ -0,0 +1,74 @@
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SharedSecret allows a Secret o be shared across namespaces.
+// Pods can mount the shared Secret or ConfigMap by adding a CSI volume to the pod specification using the
```
```suggestion
// Pods can mount the shared Secret by adding a CSI volume to the pod specification using the
```

coreydaley commented on pkg/api/sharedresource/v1alpha1/shared_secret_type.go:
```diff
@@ -0,0 +1,74 @@
+// SharedSecretSpec defines the desired state of a SharedSecret
+// +k8s:openapi-gen=true
+type SharedSecretSpec struct {
+	// resource references the backing Secret for this SharedSecret.
```
```suggestion
	// secret references the backing Secret for this SharedSecret.
```

coreydaley commented on pkg/cache/configmaps.go:
```diff
@@ -43,10 +43,10 @@ var (
 	// and configmaps if they happen to be created at roughly the same time come into play; also, if a pod with a share
 	// pointing to a configmap has been provisioned, but the the csi driver daemonset has been restarted, such timing
-	// of events where we learn of shares before their configmaps can also occur, as we attempt to rebuild the CSI driver's
+	// of events where we learn of shareConfigMaps before their configmaps can also occur, as we attempt to rebuild the CSI driver's
```
```suggestion
	// of events where we learn of sharedConfigMaps before their configmaps can also occur, as we attempt to rebuild the CSI driver's
```

coreydaley commented on pkg/cache/configmaps.go:
```diff
@@ -43,10 +43,10 @@ var (
 	configmapsWithShares = sync.Map{}
 	// sharesWaitingOnConfigmaps conversely is for when a share has been created that references a configmap, but that
-	// configmap has not been recognized by the controller; quite possibly timing events on when we learn of shares
+	// configmap has not been recognized by the controller; quite possibly timing events on when we learn of shareConfigMaps
```
```suggestion
	// configmap has not been recognized by the controller; quite possibly timing events on when we learn of sharedConfigMaps
```

coreydaley commented on pkg/cache/secrets.go:
```diff
@@ -44,10 +44,10 @@ var (
 	// and secret if they happen to be created at roughly the same time come into play; also, if a pod with a share
 	// pointing to a secret has been provisioned, but the the csi driver daemonset has been restarted, such timing
-	// of events where we learn of shares before their secrets can also occur, as we attempt to rebuild the CSI driver's
+	// of events where we learn of shareConfigMaps before their secrets can also occur, as we attempt to rebuild the CSI driver's
```
```suggestion
	// of events where we learn of sharedSecrets before their secrets can also occur, as we attempt to rebuild the CSI driver's
```

coreydaley commented on pkg/cache/secrets.go:
```diff
@@ -44,10 +44,10 @@ var (
 	secretsWithShare = sync.Map{}
 	// sharesWaitingOnSecrets conversely is for when a share has been created that references a secret, but that
-	// secret has not been recognized by the controller; quite possibly timing events on when we learn of shares
+	// secret has not been recognized by the controller; quite possibly timing events on when we learn of shareConfigMaps
```
```suggestion
	// secret has not been recognized by the controller; quite possibly timing events on when we learn of sharedSecrets
```

coreydaley commented on pkg/cache/secrets.go:
```diff
@@ -80,19 +80,19 @@ func SetSecret(kubeClient kubernetes.Interface, sharedDataKey string) error {
 		secretKey := BuildKey(br)
 		secretsWithShare.Store(secretKey, secret)
-		//NOTE: share update ranger will store share in shares sync.Map
+		//NOTE: share update ranger will store share in shareConfigMaps sync.Map
```
```suggestion
		//NOTE: share update ranger will store share in shareSecrets sync.Map
```

coreydaley commented on pkg/cache/secrets.go:
```diff
@@ -80,19 +80,19 @@ func SetSecret(kubeClient kubernetes.Interface, sharedDataKey string) error {
+		//NOTE: share update ranger will store share in shareConfigMaps sync.Map
 		// and we are supplying only this specific share to the csi driver update range callbacks.
-		shareUpdateCallbacks.Range(buildRanger(buildCallbackMap(share.Name, share)))
+		shareConfigMapsUpdateCallbacks.Range(buildRanger(buildCallbackMap(share.Name, share)))
```
```suggestion
		shareSecretsUpdateCallbacks.Range(buildRanger(buildCallbackMap(share.Name, share)))
```

coreydaley commented on pkg/cache/secrets.go:
```diff
@@ -117,7 +117,7 @@ func DelSecret(secret *corev1.Secret) {
 func RegisterSecretUpsertCallback(volID string, f func(key, value interface{}) bool) {
 	secretUpsertCallbacks.Store(volID, f)
-	// cycle through the secrets with shares, where if the share associated with the volID CSI volume mount references
+	// cycle through the secrets with shareConfigMaps, where if the share associated with the volID CSI volume mount references
```
```suggestion
	// cycle through the secrets with shareSecrets, where if the share associated with the volID CSI volume mount references
```

coreydaley commented on pkg/cache/shares.go:
```diff
@@ -43,123 +43,180 @@ variety is how we currently verify that a given pod is *STILL* allowed to access
+func AddSharedSecret(share *sharev1alpha1.SharedSecret) {
+	br := share.Spec.Secret
+	key := BuildKey(br)
+	klog.V(4).Infof("AddSharedConfigMap key %s", key)
```
```suggestion
	klog.V(4).Infof("AddSharedSecret key %s", key)
```

coreydaley commented on pkg/cache/shares.go:
```diff
@@ -43,123 +43,180 @@ variety is how we currently verify that a given pod is *STILL* allowed to access
+	shareConfigMapsDeleteCallbacks.Range(buildRanger(buildCallbackMap(share.Name, share)))
+}
+
+func DelSharedSecret(share *sharev1alpha1.SharedSecret) {
```
Missing comment for exported function

coreydaley commented on pkg/controller/controller.go:
```diff
@@ -380,60 +437,106 @@ func (c *Controller) shareEventHandler() cache.ResourceEventHandlerFuncs {
-			if err := c.syncShare(event); err != nil {
-				c.shareWorkqueue.AddRateLimited(obj)
+			if err := c.syncSharedSecret(event); err != nil {
+				c.sharedConfigMapWorkqueue.AddRateLimited(obj)
```
```suggestion
				c.sharedSecretWorkqueue.AddRateLimited(obj)
```

coreydaley commented on pkg/controller/controller.go:
```diff
@@ -380,60 +437,106 @@ func (c *Controller) shareEventHandler() cache.ResourceEventHandlerFuncs {
+				c.sharedConfigMapWorkqueue.AddRateLimited(obj)
 			} else {
-				c.shareWorkqueue.Forget(obj)
+				c.sharedConfigMapWorkqueue.Forget(obj)
```
```suggestion
				c.sharedSecretWorkqueue.Forget(obj)
```

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -187,7 +187,7 @@ func commonUpsertRanger(obj runtime.Object, podPath, filter string, key, value i
 	// host filesystem either being full or compromised in some long running fashion, so tight-loop retry, like we
 	// *could* do here as a result will typically prove fruitless.
-	// Then, the controller relist will result in going through the secrets/configmaps we share, so
+	// Then, the controller relist will result in going through the secrets/configmaps we secretShare, so
```
```suggestion
	// Then, the controller relist will result in going through the secrets/configmaps we share, so
```

I think your search/replace got a bit overzealous here :)

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -241,8 +241,8 @@ func commonOSRemove(dir string) error {
 		}
-		// since we do not support mounting on existing content, a dir can only mean a share
-		// has been mounted as a separate dir in our share, so skip
+		// since we do not support mounting on existing content, a dir can only mean a secretShare
```
```suggestion
		// since we do not support mounting on existing content, a dir can only mean a share
```

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -241,8 +241,8 @@ func commonOSRemove(dir string) error {
-		// since we do not support mounting on existing content, a dir can only mean a share
-		// has been mounted as a separate dir in our share, so skip
+		// since we do not support mounting on existing content, a dir can only mean a secretShare
+		// has been mounted as a separate dir in our secretShare, so skip
```
```suggestion
		// has been mounted as a separate dir in our share, so skip
```

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -267,21 +267,21 @@ func shareDeleteRanger(hp *hostPath, key interface{}) bool {
 			}
 			volID = hpv.GetVolID()
-			// deleting the share effectively deletes permission to the
+			// deleting the secretShare effectively deletes permission to the
```
```suggestion
			// deleting the share effectively deletes permission to the
```

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -267,21 +267,21 @@ func shareDeleteRanger(hp *hostPath, key interface{}) bool {
+			// deleting the secretShare effectively deletes permission to the
 			// data so we set the allowed bit to false; this will have bearing
-			// if the share is added again at a later date and the associated
+			// if the secretShare is added again at a later date and the associated
```
```suggestion
			// if the share is added again at a later date and the associated
```

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -292,59 +292,86 @@ func shareDeleteRanger(hp *hostPath, key interface{}) bool {
 		}
 		// we just delete the associated data from the previously provisioned volume;
-		// we don't delete the volume in case the share is added back
+		// we don't delete the volume in case the secretShare is added back
```
```suggestion
		// we don't delete the volume in case the share is added back
```

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -292,59 +292,86 @@ func shareDeleteRanger(hp *hostPath, key interface{}) bool {
 		storeVolMapToDisk()
 	}
-	klog.V(4).Infof("share delete ranger returning share id %s", shareId)
+	klog.V(4).Infof("shareDeleteRanger returning secretShare id %s", shareId)
```
```suggestion
	klog.V(4).Infof("shareDeleteRanger returning share id %s", shareId)
```

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -292,59 +292,86 @@ func shareDeleteRanger(hp *hostPath, key interface{}) bool {
+	sharedSecret, sok := value.(*sharev1alpha1.SharedSecret)
+	sharedConfigMap, cmok := value.(*sharev1alpha1.SharedConfigMap)
+	if !sok && !cmok {
+		klog.Warningf("unknown shareUpdateRanger key %s type %#v", shareId, value)
```
```suggestion
		klog.Warningf("unknown shareUpdateRanger key %q type %q", shareId, value)
```

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -292,59 +292,86 @@ func shareDeleteRanger(hp *hostPath, key interface{}) bool {
 	ranger := func(key, value interface{}) bool {
 		hpv, _ = value.(*hostPathVolume)
-		klog.V(4).Infof("share update ranger id %s share name %s type %s hpv ranger", shareId, share.Name, share.Spec.Resource.Type)
+		klog.V(4).Infof("shareUpdateRanger id %s hpv ranger", shareId)
```
```suggestion
		klog.V(4).Infof("shareUpdateRanger id %q hpv ranger", shareId)
```

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -292,59 +292,86 @@ func shareDeleteRanger(hp *hostPath, key interface{}) bool {
+				shareName = sharedConfigMap.Name
+			}
+
+			klog.V(4).Infof("shareUpdateRanger id %s secretShare name %s type %s hpv ranger found volume %s", shareId, shareName, shareType, hpv.GetVolID())
```
```suggestion
			klog.V(4).Infof("shareUpdateRanger id %s share name %s type %s hpv ranger found volume %s", shareId, shareName, shareType, hpv.GetVolID())
```

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -427,7 +454,7 @@ func shareUpdateRanger(key, value interface{}) bool {
 	}
 
-	klog.V(4).Infof("share update ranger id %s share name %s type %s version %s returning", shareId, share.Name, share.Spec.Resource.Type, share.ResourceVersion)
+	klog.V(4).Infof("secretShare update ranger id %s secretShare name %s type %s version %s returning", shareId, shareName, shareType, shareRV)
```
```suggestion
	klog.V(4).Infof("share update ranger id %s share name %s type %s version %s returning", shareId, shareName, shareType, shareRV)
```

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -535,7 +562,7 @@ func mapBackingResourceToPod(hpv *hostPathVolume) error {
 		objcache.RegisterSecretDeleteCallback(hpv.GetVolID(), deleteRangerSec)
 	default:
-		return fmt.Errorf("invalid share backing resource kind %s", hpv.GetSharedDataKind())
+		return fmt.Errorf("invalid secretShare backing resource kind %s", hpv.GetSharedDataKind())
```
```suggestion
		return fmt.Errorf("invalid share backing resource kind %s", hpv.GetSharedDataKind())
```

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -551,7 +578,7 @@ func (hp *hostPath) updateObjCache(hpv *hostPathVolume) error {
 		return objcache.SetSecret(hp.kubeClient, key)
 	default:
-		return fmt.Errorf("invalid share backing resource kind %s", kind)
+		return fmt.Errorf("invalid secretShare backing resource kind %s", kind)
```
```suggestion
		return fmt.Errorf("invalid share backing resource kind %s", kind)
```

coreydaley commented on pkg/hostpath/hostpath.go:
```diff
@@ -684,11 +731,13 @@ func (hp *hostPath) deleteHostpathVolume(volID string) error {
 	objcache.UnregisterSecretUpsertCallback(volID)
+	objcache.UnregisterConfigMapUpsertCallback(volID)
 	objcache.UnregisterSecretDeleteCallback(volID)
 	objcache.UnregisterConfigMapDeleteCallback(volID)
-	objcache.UnregisterConfigMapUpsertCallback(volID)
-	objcache.UnregisterShareDeleteCallback(volID)
-	objcache.UnregisterShareUpdateCallback(volID)
+	objcache.UnregisterSharedConfigMapDeleteCallback(volID)
+	objcache.UnregisterSharedSecretDeleteCallback(volID)
+	objcache.UnregsiterSharedSecretsUpdateCallback(volID)
+	objcache.UnregisterSharedConfigMapUpdateCallback(volID)
```
I feel like you are missing a couple of `Unregister<thing>`here, you might want to double-check

gabemontero replied:
double checked and I think I have all of them ... the ordering could be improved ... I'll include that in my next push

coreydaley commented on pkg/hostpath/hpv.go:
```diff
@@ -27,7 +27,7 @@ type hostPathVolume struct {
 	Allowed             bool       `json:"allowed"`
 	ReadOnly            bool       `json:"readOnly"`
-	// hpv's can be accessed/modified by both the share events and the configmap/secret events; to prevent data races
+	// hpv's can be accessed/modified by both the secretShare events and the configmap/secret events; to prevent data races
```
```suggestion
	// hpv's can be accessed/modified by both the share events and the configmap/secret events; to prevent data races
```

coreydaley commented on pkg/hostpath/mount.go:
```diff
@@ -32,7 +32,7 @@ type FileSystemMounter interface {
 // and corroboration with OpenShift storage SMEs, a separate tmpfs per pod
 // - ensures the kubelet will handle SELinux for us. It will relabel the volume in "the right way" just for the pod
-// - otherwise, if pods share the same host dir, all sorts of warnings from the SMEs
+// - otherwise, if pods secretShare the same host dir, all sorts of warnings from the SMEs
```
```suggestion
// - otherwise, if pods share the same host dir, all sorts of warnings from the SMEs
```

coreydaley commented on pkg/hostpath/nodeserver.go:
```diff
@@ -68,43 +67,75 @@ func getPodDetails(volumeContext map[string]string) (string, string, string, str
+	secretShareName, sok := req.GetVolumeContext()[SharedSecretShareKey]
+	if (!cmok && !sok) || (len(strings.TrimSpace(configMapShareName)) == 0 && len(strings.TrimSpace(secretShareName)) == 0) {
+		return nil, nil, status.Errorf(codes.InvalidArgument,
+			"the csi driver reference is missing the volumeAttribute '%s' and '%s'", SharedSecretShareKey, SharedConfigMapShareKey)
```
```suggestion
			"the csi driver reference is missing the volumeAttribute %q and %q", SharedSecretShareKey, SharedConfigMapShareKey)
```

coreydaley commented on pkg/hostpath/nodeserver.go:
```diff
@@ -68,43 +67,75 @@ func getPodDetails(volumeContext map[string]string) (string, string, string, str
-			"the csi driver volumeAttribute 'share' reference had an error: %s", err.Error())
+	if (cmok && sok) || (len(strings.TrimSpace(configMapShareName)) > 0 && len(strings.TrimSpace(secretShareName)) > 0) {
+		return nil, nil, status.Errorf(codes.InvalidArgument,
+			"a single volume cannot support both a SharedConfigMap reference (%s) and SharedSecret reference (%s)",
```
```suggestion
			"a single volume cannot support both a SharedConfigMap reference %q and SharedSecret reference %q",
```

coreydaley commented on pkg/hostpath/nodeserver.go:
```diff
@@ -68,43 +67,75 @@ func getPodDetails(volumeContext map[string]string) (string, string, string, str
+		cmShare, err = client.GetListers().SharedConfigMaps.Get(configMapShareName)
+		if err != nil {
+			return nil, nil, status.Errorf(codes.InvalidArgument,
+				"the csi driver volumeAttribute '%s' reference had an error: %s", configMapShareName, err.Error())
```
```suggestion
				"the csi driver volumeAttribute %q reference had an error: %s", configMapShareName, err.Error())
```

coreydaley commented on pkg/hostpath/nodeserver.go:
```diff
@@ -68,43 +67,75 @@ func getPodDetails(volumeContext map[string]string) (string, string, string, str
+		sShare, err = client.GetListers().SharedSecrets.Get(secretShareName)
+		if err != nil {
+			return nil, nil, status.Errorf(codes.InvalidArgument,
+				"the csi driver volumeAttribute '%s' reference had an error: %s", secretShareName, err.Error())
```
```suggestion
				"the csi driver volumeAttribute %q reference had an error: %s", secretShareName, err.Error())
```

coreydaley commented on pkg/hostpath/nodeserver.go:
```diff
@@ -68,43 +67,75 @@ func getPodDetails(volumeContext map[string]string) (string, string, string, str
+	if cmShare != nil {
+		if len(strings.TrimSpace(cmShare.Spec.ConfigMap.Namespace)) == 0 {
+			return nil, nil, status.Errorf(codes.InvalidArgument,
+				"the SharedConfigMap %s backing resource namespace needs to be set", configMapShareName)
```
```suggestion
				"the SharedConfigMap %q backing resource namespace needs to be set", configMapShareName)
```

coreydaley commented on pkg/hostpath/nodeserver.go:
```diff
@@ -68,43 +67,75 @@ func getPodDetails(volumeContext map[string]string) (string, string, string, str
+		}
+		if len(strings.TrimSpace(cmShare.Spec.ConfigMap.Name)) == 0 {
+			return nil, nil, status.Errorf(codes.InvalidArgument,
+				"the SharedConfigMap %s backing resource name needs to be set", configMapShareName)
```
```suggestion
				"the SharedConfigMap %q backing resource name needs to be set", configMapShareName)
```

coreydaley commented on pkg/hostpath/nodeserver.go:
```diff
@@ -68,43 +67,75 @@ func getPodDetails(volumeContext map[string]string) (string, string, string, str
+		kind = sharev1alpha1.ResourceReferenceTypeSecret
+		if len(strings.TrimSpace(sShare.Spec.Secret.Namespace)) == 0 {
+			return nil, nil, status.Errorf(codes.InvalidArgument,
+				"the SharedSecret %s backing resource namespace needs to be set", secretShareName)
```
```suggestion
				"the SharedSecret %q backing resource namespace needs to be set", secretShareName)
```

coreydaley commented on pkg/hostpath/nodeserver.go:
```diff
@@ -68,43 +67,75 @@ func getPodDetails(volumeContext map[string]string) (string, string, string, str
+		}
+		if len(strings.TrimSpace(sShare.Spec.Secret.Name)) == 0 {
+			return nil, nil, status.Errorf(codes.InvalidArgument,
+				"the SharedSecret %s backing resource name needs to be set", secretShareName)
```
```suggestion
				"the SharedSecret %q backing resource name needs to be set", secretShareName)
```

gabemontero commented on pkg/controller/controller.go:
```diff
@@ -127,19 +134,23 @@ func (c *Controller) Run(stopCh <-chan struct{}) error {
 
 	c.informerFactory.Start(stopCh)
-	c.shareInformerFactory.Start(stopCh)
+	c.sharedConfigMapInformerFactory.Start(stopCh)
```
yep I missed it - thanks

gabemontero commented:
ok @coreydaley I believe I have addressed the latest set of comments in 2 "core-3" commits (where I squashed the previous review commits

unit tests pass for me locally but I'm still waiting on my cluster to come up to try the e2e's

I'll run those locally when the cluster comes up if I see any failures here in the PR

gabemontero commented:
> Should we update the corresponding enhancement proposal with this change ?

yes I'm including that as one of the to-do's for BUILD-353

just getting the code ready first

gabemontero commented:
aws error with slow

/test e2e-aws-csi-driver-slow

gabemontero commented:
ok the e2e's are passing for me locally ... would anticipate the slow e2e passing if it gets past unrelated flakes

gabemontero commented:
and the e2e's are green as well @coreydaley 

assuming there is nothing else, and @adambkaplan deferral to us in scrum/office hours today, I'll squash and then we can lgtm / merge

And you can start mapping the api changes back to https://github.com/openshift/api/pull/979

coreydaley reviewed (changes requested):
I think these are the last two issues.
If you want to squash your commits and then I think we can get this merged and I can proceed with the openshift/api stuff.

gabemontero commented:
corrections pulled in and pushed, commits squashed @coreydaley thanks !!

coreydaley commented:
/lgtm
